### DISCUSSION TITLE
questions on tests

### DISCUSSION THREAD

**Discussion by pengguanya** [insightsengineering/rbmi#366] [Q&A]
@gowerc 
__Q1__:

in which file the `.test_print` object is created and how it is sourced? I need to update it to make the test pass for new structure
https://github.com/insightsengineering/rbmi/blob/a45df71eeff8b3a134523a5c73e47aa986953ff2/tests/testthat/test-print.R#L7-L16

__Q2__:

how to switch on full test?
https://github.com/insightsengineering/rbmi/blob/a45df71eeff8b3a134523a5c73e47aa986953ff2/tests/testthat/test-mcmc.R#L328

__Q3__
related to __Q1__. I guess the snapshot for test-print needs to be reviewed and updated since the format of pool has been changed now. What is the work-flow to update the snapshot?
<img width="457" alt="image" src="https://user-images.githubusercontent.com/4429650/173830883-c81755cd-d08a-48a6-ae75-88fb3699d728.png">

**Comment by gowerc** [comment]
Hey @pengguanya ,  

The print datasets are defined [here](https://github.com/insightsengineering/rbmi/blob/main/data-raw/create_print_test_data.R)

This script creates the the relevant objects and stores them in `R/sysdata.rda` (its a binary file). This file is often used for loading pre-derived data objects (see [here](https://r-pkgs.org/data.html) for more details)
